### PR TITLE
chore: add clang-tidy

### PR DESCRIPTION
**Pull request by gocarlos**
# WHAT
adds clang-tidy capabilities to the cmake script

# WHY
Some code parts could be improved/modernized, clang-tidy helps with this task by showing problematic parts e.g. `if` statemenets with wrong indentation, or without brakets etc.

### CODE REVIEW

**Comment by gocarlos** [comment]
@jimsch please review

**Comment by gocarlos** [comment]
you will see the CI in the `actions` tab here in github after merging this, currently only in my fork: https://github.com/gocarlos/COSE-C/actions/runs/82682570

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        USE_MBEDTLS: ["ON", "OFF"]
+        USE_CONTEXT: ["ON", "OFF"]
+        CMAKE_OPTIONS:
+          [
+            "-DCOSE_C_INCLUDE_ENCRYPT=ON -DCOSE_C_INCLUDE_encrypt0=OFF -DCOSE_C_INCLUDE_MAC0=OFF -DCOSE_C_INCLUDE_MAC=OFF -DCOSE_C_INCLUDE_SIGN=OFF -DCOSE_C_INCLUDE_SIGN1=OFF",
+            "-DCOSE_C_INCLUDE_ENCRYPT=OFF -DCOSE_C_INCLUDE_encrypt0=ON -DCOSE_C_INCLUDE_MAC0=OFF -DCOSE_C_INCLUDE_MAC=OFF -DCOSE_C_INCLUDE_SIGN=OFF -DCOSE_C_INCLUDE_SIGN1=OFF",
```
Comment (jimsch) [inline .github/workflows/main.yml]: COSE_C_INCLUDE_encrypt0 should be capitalized.

File: CMakeLists.txt [hunk 2]
```diff
@@ -141,6 +140,24 @@ if(COSE_C_BUILD_DOCS)
...
+
+if(COSE_C_RUN_CLANG_TIDY)
+  find_program(
+    CLANG_TIDY_EXE
+    NAMES "clang-tidy"
+    DOC "/usr/bin/clang-tidy")
+  if(NOT CLANG_TIDY_EXE)
+    message(WARNING "clang-tidy not found.")
+  else()
+    message(STATUS "clang-tidy found: ${CLANG_TIDY_EXE}")
```
Comment (jimsch) [inline CMakeLists.txt]: This message seems redundant as you are dumping out in the epilog list of objects.  No action required.

**Review by jimsch — CHANGES_REQUESTED** [review]
When I look at the pointer above, I am not seeing the tests being run.   Is that because they aren't run in the artifacts but are run elsewhere or because they are just not being run?

**Comment by gocarlos** [comment]
To keep MRs as small as possible: this one does not run tests, we run them already with travis and appveyer....

Here we only do static analysis with clang-tidy in github actions...

We can port tests to here in the future..

**Pull request merged**